lib/bolt/cli.rb

```diff
@@ -919,9 +941,17 @@ def install_modules(puppetfile_path, config, moduledir, modules)
 
     # Loads a Puppetfile and installs its modules.
     #
-    def install_puppetfile(config, puppetfile, moduledir)
+    def install_puppetfile(config, puppetfile, moduledir, modules)
       require 'bolt/puppetfile/installer'
 
+      if modules
```

Comment:
This seems like it'd make more sense directly in the `when 'puppetfile'` clause before the case statement so that this is less prone to getting raised in situations where we're not running a puppetfile subcommand.